src/main/scala/viper/gobra/frontend/InformativeErrorListener.scala

```
@@ -0,0 +1,246 @@
+    e match {
+      case exception: FailedPredicateException => analyzeFailedPredicate(context, exception)
+      case exception: InputMismatchException => analyzeInputMismatch(context, exception)
+      case exception: NoViableAltException => analyzeNoViable(context, exception)
+      case null => context.msg match {
+        case extraneous() => analyzeExtraneous(context)
+        case missing() => DefaultErrorType()
+      }
+      case _ => DefaultErrorType()
+    }
+  }
+
+
+  def analyzeFailedPredicate(implicit context: ParserErrorContext, exception: FailedPredicateException): ErrorType = {
+    val parser = context.recognizer
+    parser.getContext match {
+      // One example of a known pattern: Parser reads ':=' when expecting the end of statement: The user probably
+      // used ':=' instead of '='
+      case _ : GobraParser.EosContext => {
+        context.offendingSymbol.getType match {
+          case GobraParser.DECLARE_ASSIGN => GotAssignErrorType()
+          case _ => DefaultFailedEOS()
+        }
+      }
+      case _ => DefaultErrorType()
+    }
+  }
+
+  def analyzeExtraneous(implicit context: ParserErrorContext): ErrorType = {
+    (context.offendingSymbol.getType, context.recognizer.getContext) match {
+      case (GobraParser.DECLARE_ASSIGN, _ : TypeSpecContext) => GotAssignErrorType()
+      case (GobraParser.R_BRACKET, expr : ExpressionContext) if expr.parent.isInstanceOf[CapContext] => SliceMissingIndex(3)
+      case _ => DefaultExtraneous()
+    }
+  }
+
+  def analyzeInputMismatch(implicit context: ParserErrorContext, exception: InputMismatchException): ErrorType = {
+    (context.offendingSymbol.getType, context.recognizer.getContext) match {
+      case (Token.EOF, _) => DefaultMismatch()
+      case (s, i : ImplementationProofContext) if context.recognizer.getExpectedTokens == IntervalSet.of(GobraParser.IMPL)=> {
+        IgnoreError()
```

I assume this "IgnoreError" does not actually mean to ingore the error? Otherwise, why do we ignore an error?

---

did this vanish? I cannot find it anymore.